src/connectors/sort-by/__tests__/connectSortBy-test.js

```diff
@@ -257,6 +258,169 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/sort-by/js/
     }
   });
 
+  describe('getRenderState', () => {
+    test('returns the render state', () => {
+      const renderFn = jest.fn();
+      const unmountFn = jest.fn();
+      const createSortBy = connectSortBy(renderFn, unmountFn);
+      const sortBy = createSortBy({
+        items: [
+          { label: 'default', value: 'index_default' },
+          { label: 'asc', value: 'index_asc' },
+          { label: 'desc', value: 'index_desc' },
+        ],
+      });
+      const helper = algoliasearchHelper(
+        createSearchClient(),
+        'index_default',
+        {
+          index: 'index_default',
+        }
+      );
+
+      const renderState1 = sortBy.getRenderState(
+        {
+          sortBy: {},
+        },
+        createInitOptions({ helper })
+      );
+
+      expect(renderState1.sortBy).toEqual({
+        currentRefinement: 'index_default',
+        refine: undefined,
+        hasNoResults: true,
+        options: [
+          { label: 'default', value: 'index_default' },
+          { label: 'asc', value: 'index_asc' },
+          { label: 'desc', value: 'index_desc' },
+        ],
+        widgetParams: {
+          items: [
+            { label: 'default', value: 'index_default' },
+            { label: 'asc', value: 'index_asc' },
+            { label: 'desc', value: 'index_desc' },
+          ],
+        },
+      });
+
+      sortBy.init(createInitOptions({ helper }));
+      sortBy.getWidgetRenderState({ helper }).refine('index_desc');
```

Comment:
We used to have this pattern in tests:

```js
      const { refine } = renderFn.mock.calls[renderFn.mock.calls.length - 1][0];
      refine('index_desc');
```

It reads what's rendered from `init` call and get `refine` from there.
However, getting `refine` from `getWidgetRenderState()` also looks okay to me.

Just leaving this comment for Clément's information, and to see if others think differently.

Reply:
Thanks for the heads up!

You're right, I think I saw it in an other connector but now that we have `getWidgetRenderState` it might be easier to understand (at least for me 🤔 ).

Reply:
yes, I think using getWidgetRenderState is the way to go for new tests I think. It's okay if older tests still use `refine` from the calls, since the implementation should be the same anyway 👍

Reply:
Since render state is now public API it makes sense to use it—and it's such a relief!